src/gamepad/controller.cpp

```diff
@@ -1,8 +1,44 @@
 #include "gamepad/controller.hpp"
 #include "gamepad/todo.hpp"
+#include "pros/misc.h"
+
+#include <new>
 
 namespace Gamepad {
 
+union ControllerBuf {
+        // this is here to allow us to control when we initialize the Controller
+        char c;
+        Controller as_controller;
+
+        // empty destructor, the Controller must be manually destroyed by ControllerInit
+        ~ControllerBuf() {}
+};
+
+static uint32_t nifty_counter; // zero initialized at load time
```

Comment:
while it is pretty nifty, we should probably rename it to just `counter`. Unless its a technical term im not aware of, in which case feel free to mock me